### PR TITLE
JENKINS-30115 Allow downstream jobs to attach GitHub status messages

### PR DESCRIPTION
**Pull request by kevinsuwala**
- Added support for GitHub contexts and messages to be added from downstream jobs
- This pull request is related to this issue: https://issues.jenkins-ci.org/browse/JENKINS-30115

### CODE REVIEW

**Comment by emanuelez** [comment]
Nice! @DavidTanner are you up for a review? :)

**Comment by Puneeth-n** [comment]
When will this be merged? :) So looking forward to this feature.

**Comment by Puneeth-n** [comment]
@kevinsuwala I compiled your branch and updated my jenkins' ghprb plugin. How do I configure downstream jobs to send contexts to upstream jobs?

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbBuilds.java [hunk 1]
```diff
@@ -59,6 +59,7 @@ public void build(GhprbPullRequest pr, GHUser triggerSender, String commentBody)
 
 
     public void onStarted(AbstractBuild<?, ?> build, TaskListener listener) {
+        GhprbCustomStatusRepoPasser.addRepo(build.getFullDisplayName(), repo.getGitHubRepo());
```
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbBuilds.java]: So right now this is a memory leak.  Can we remove the repo from the list when the build is completed, or when is it safe to be removed?

**Comment by DavidTanner** [comment]
Please check my pull request against your branch.

**Comment by DavidTanner** [comment]
Closed for #180

**Pull request closed without merging**